plugins/modules/certificate.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: certificate
+
+author:
+  - Domen Dobnikar (@domen_dobnikar)
+short_description: Handles cluster SSL certificates.
+description:
+  - Can upload and verify SSL certificates.
+version_added: 1.1.0
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+seealso: []
+options:
+  certificate:
+    description:
+      - File location containing the X.509 PEM-encode certificate.
+    type: str
+    required: true
+  private_key:
+    description:
+      - File location containing the RSA PEM-encoded private key.
+    type: str
+    required: true
+notes:
+  - C(check_mode) is not supported.
+"""
+
+EXAMPLES = r"""
+- name: Upload new certificate
+  scale_computing.hypercore.certificate:
+    private_key: "{{ lookup('file', 'private_key.pem') }}"
+    certificate: "{{ lookup('file', 'scale_cert.cer') }}"
+"""
+
+RETURN = r"""
+record:
+  description:
+    - Certificate record.
+  returned: success
+  type: dict
+  sample:
+    certificate: this_is_the_certificate
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors
+from ..module_utils.utils import is_changed
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.typed_classes import (
+    TypedDiff,
+    TypedTaskTag,
+    TypedCertificateToAnsible,
+)
+from ..module_utils.task_tag import TaskTag
+
+from typing import Union, Tuple
+import ssl
+from time import sleep
+
+
+def get_certificate(module: AnsibleModule) -> str:
+    host = (
+        module.params["cluster_instance"]["host"]
+        .replace("https://", "")
+        .replace("http://", "")
+    )
+    cert = ssl.get_server_certificate((host, 443))
+    return cert
+
+
+def upload_cert(module: AnsibleModule, rest_client: RestClient) -> TypedTaskTag:
+    payload = dict(
+        certificate=module.params["certificate"],
+        privateKey=module.params["private_key"],
+    )
+    response = rest_client.create_record("/rest/v1/Certificate", payload, False)
+    return response
+
+
+def ensure_present(
+    module: AnsibleModule,
+    rest_client: RestClient,
+) -> Tuple[bool, Union[TypedCertificateToAnsible, None], TypedDiff]:
+    before: TypedCertificateToAnsible = dict(certificate=get_certificate(module))
+    # Don't upload if not necessary
+    if before["certificate"].replace("\n", "") == module.params["certificate"].replace(
+        "\n", ""
+    ):
+        return False, None, dict(before=None, after=None)
```

Comment:
Imagine you upload certA + keyB, later you figure out mistake, and try to fix by uploading certA + keyA.
Module will not be helpful, because it cannot know key is diffrent.

I think it is better to drop idempotence here, and always POST to server, always return changed=True.
We do same also with password for SMTP config, and password/secret for OIDC config.

Reply:
Yes good point this can definitely be an issue.
Will do.